lib/logstash/inputs/lumberjack.rb

```diff
@@ -73,14 +73,15 @@ def run(output_queue)
       if @circuit_breaker.closed?
         connection = @lumberjack.accept # Blocking call that creates a new connection
 
-        invoke(connection, @codec.clone) do |_codec, line, fields|
-          _codec.decode(line) do |event|
+        invoke(connection, @codec.clone) do |code, _codec, fields|
+          handler = code == :json ? json_event : data_event
```

Comment:
need to improve unit tests then. this was unnoticed due to tests still being successful.